contrib/platform/test/com/sun/jna/platform/win32/MsiTest.java

```diff
@@ -33,52 +33,57 @@ public static void main(String[] args) {
 
     public void testMsiEnumComponents() {
         char[] componentBuffer = new char[40];
-        assertTrue(W32Errors.ERROR_SUCCESS == Msi.INSTANCE.MsiEnumComponents(new WinDef.DWORD(0), componentBuffer));
+        assertEqualsWithWin32Error(W32Errors.ERROR_SUCCESS, Msi.INSTANCE.MsiEnumComponents(new WinDef.DWORD(0), componentBuffer));
         String component = new String(componentBuffer).trim();
         assertTrue(component.length() > 0);
     }
 
     public void testMsiGetProductCodeW() {
         char[] componentBuffer = new char[40];
-        assertTrue(W32Errors.ERROR_SUCCESS == Msi.INSTANCE.MsiEnumComponents(new WinDef.DWORD(0), componentBuffer));
+        assertEqualsWithWin32Error(W32Errors.ERROR_SUCCESS, Msi.INSTANCE.MsiEnumComponents(new WinDef.DWORD(0), componentBuffer));
         String component = new String(componentBuffer).trim();
 
         char[] productBuffer = new char[40];
-        assertTrue(W32Errors.ERROR_SUCCESS == Msi.INSTANCE.MsiGetProductCode(component, productBuffer));
+        assertEqualsWithWin32Error(W32Errors.ERROR_SUCCESS, Msi.INSTANCE.MsiGetProductCode(component, productBuffer));
 
         String product = new String(productBuffer).trim();
         assertTrue(product.length() > 0);
     }
 
     public void testMsiLocateComponentW() {
         char[] componentBuffer = new char[40];
-        assertTrue(W32Errors.ERROR_SUCCESS == Msi.INSTANCE.MsiEnumComponents(new WinDef.DWORD(0), componentBuffer));
+        assertEqualsWithWin32Error(W32Errors.ERROR_SUCCESS, Msi.INSTANCE.MsiEnumComponents(new WinDef.DWORD(0), componentBuffer));
         String component = new String(componentBuffer).trim();
 
         char[] pathBuffer = new char[WinDef.MAX_PATH];
         IntByReference pathBufferSize = new IntByReference(pathBuffer.length);
-        Msi.INSTANCE.MsiLocateComponent(component, pathBuffer, pathBufferSize);
+        System.out.println("InstallState: " + Msi.INSTANCE.MsiLocateComponent(component, pathBuffer, pathBufferSize));
```

Comment:
Updated in PR https://github.com/java-native-access/jna/pull/1224.
Can't do an assert since I do not know what values that are allowed to still continue with the rest of the function. In an case, follow-up discussion should be in https://github.com/java-native-access/jna/pull/1224.